Support react 16(react-native 0.43) , findNodeHandle, FlatList and SectionList

flyskywhy opened the pull request:
react 16:
(Chinese version comment of commit 8c543ba) react-native 0.43 对应的 react 的版本是 16.0.0-alpha.6 ，由于 react@16.0.0-alpha.6 中的 ReactBaseClasses.js 中的 setState 调用 enqueueSetState() 的参数与 react-dom@15.4.2 中
    的 ReactUpdateQueue.js 中的 enqueueSetState() 定义配合不上，因此需要升级 react-dom 以使 Web 版的 Navigator 的页面 route 跳转再次能用，这实际上就是让 setState(partialState, callback) 中的 callback 再次能用。而之所以只升级成 react-dom@16.0.0-alpha.2 而非 react-dom@16.0.0-alpha.6 是因为 ReactAnimated 所用的第三方组件 animated 现在还不支持 [React v16 Support](https://github.com/animatedjs/animated/issues/63) 而使得从 react-dom@16.0.0-alpha.3 到  react-dom@16.0.0-alpha.6 都会导致崩溃，所以现在先暂时锁定在唯一能用的 react-dom@16.0.0-alpha.2


另：@yuanyan 后续是否考虑进行类似 https://github.com/necolas/react-native-web/commit/941c6284455bb80674ebe15d155547bfbba1bc2b 这样的去除一些对 react-dom 依赖的操作？就像 necolas 在  [React v16 Support](https://github.com/animatedjs/animated/issues/63) 中建议 https://github.com/animatedjs/animated/blob/master/src/targets/react-dom.js 中的 `var CSSPropertyOperations = require('react-dom/lib/CSSPropertyOperations');` 语句去除对 react-dom 的依赖那样


https://github.com/react-community/react-navigation:
work with react-navigation recommended by react-native@0.44.0(it remove navigator) when
use
    `import { StackNavigator } from 'react-navigation/src/react-navigation.js'`
not
    `import { StackNavigator } from 'react-navigation'; // it will use react-navigation.web.js instead`

and replace `PlatformHelpers` with `PlatformHelpers.native` in react-navigation/src/createNavigationContainer.js


findNodeHandle:
FlatList need it

FlatList and SectionList:
ported from react-native@0.43.3

flyskywhy committed:
work with react@16.0.0-alpha.6 from react-native@0.43.0, and the reason
we set react-dom@16.0.0-alpha.2 but not react-dom@16.0.0-alpha.6 is:

1. react@16.0.0-alpha.6 -> ReactBaseClasses.js -> setState -> enqueueSetState()
is not compatible with the parameters defined in react-dom@15.4.2 ->
ReactUpdateQueue.js -> enqueueSetState(), so we need upgrade react-dom, othewise
navigator can't route to new page because it use setState;

2. ReactAnimated use animated, but animated has not suppor React v16 for now, see
https://github.com/animatedjs/animated/issues/63 , and react-dom@16.0.0-alpha.3
to react-dom@16.0.0-alpha.6 will crash the animated but accurate
react-dom@16.0.0-alpha.2 is good.

flyskywhy committed:
work with react-navigation recommended by react-native@0.44.0 when

use
    import { StackNavigator } from 'react-navigation/src/react-navigation.js'
not
    import { StackNavigator } from 'react-navigation'; // it will use react-navigation.web.js instead

and replace PlatformHelpers with PlatformHelpers.native in react-navigation/src/createNavigationContainer.js

flyskywhy committed:
FlatList(step 1): just copy from react-native@0.43.3

flyskywhy committed:
findNodeHandle: added, ref to https://github.com/necolas/react-native-web/blob/master/src/modules/findNodeHandle/index.js

flyskywhy committed:
FlatList(step 2): worked well

flyskywhy committed:
NetInfo: added, ref to https://github.com/necolas/react-native-web/blob/master/src/apis/NetInfo/index.js

cpunion commented:
@flyskywhy 看起来这个库放弃维护了，有动力接手继续吗？

flyskywhy committed:
NetInfo: add origin Copyright

flyskywhy committed:
FlatList(step 3): worked well when scroll

flyskywhy committed:
FlatList(step 4): refactor faster when scroll

flyskywhy committed:
support scrollEnabled prop for ViewPager

flyskywhy committed:
support Animated.ScrollView

flyskywhy committed:
support scrollEnabled prop for ScrollView

cpunion committed:
Remove dependency of react-dom/lib/CSSPropertyOperations

cpunion committed:
Import PropTypes from prop-types

flyskywhy committed:
Merge pull request #1 from cpunion/flyskywhy

More react 16 compatible

@cpunion ，谢谢信任 :smile:

flyskywhy commented:
@cpunion ，谢谢信任 :smile: ，您的 PR 已合并

flyskywhy commented:
@fov42550564 ，issue 暂时还是先集中在这边吧，我也会时不时过来看看的

cpunion committed:
Convert React.createClass to ES6 class

cpunion committed:
Use Array.prototype.find instead of array-find-index

cpunion committed:
ScrollView scrolls with animation

cpunion committed:
Fix first rendering of transform

cpunion committed:
Make ScrollView.defaultProps.scrollEnabled default true

flyskywhy committed:
Merge pull request #2 from cpunion/flyskywhy

Convert React.createClass to ES6 class

flyskywhy committed:
Merge pull request #3 from cpunion/remove-array-find-index

Remove array-find-index

fov42550564 commented:
@flyskywhy   FlatList不能下拉刷新，设置refreshing={true}后，activityindicato也不能显示出来，onRefresh={()=>{console.log("========");}}函数也不执行，你测试的时候遇到这个问题了吗，另外，老版本的ListView设置refreshControl也不能下来刷新，是不是需要采用第三发的List？

cpunion commented:
咱们 fork 一个共同维护吧，以这边的情况看，不但基本修复合并不了，以后发展也没指望了。

fov42550564 commented:
叫 @flyskywhy  fork 吧

flyskywhy commented:
@fov42550564 不能下拉刷新这个问题我也发现了，花了点时间但没解决，后来考虑到还要禁止微信/支付宝上自带的下拉提示（网页有 xx 提供），所以最后没有去解决下拉刷新问题，而是在 APP 中有列表的地方 Platform.OS === 'web' 时就放置一个刷新按钮来 workaround 的。

要不请 @cpunion @fov42550564  两位共同在我的 fork https://github.com/flyskywhy/react-web 中维护？我会尽量及时查看 diff 、实际测试然后合并 PR 的，谢谢 :smile:

cpunion commented:
@flyskywhy 我之前把 react-16 相关的提给你了，其它的提在这边 https://github.com/taobaofed/react-web/pull/259 #261。以后都提到你那边吧，我还有几个小 fix 的没提。😄

fov42550564 commented:
https://github.com/flyskywhy/react-web中好像不能提交Issue

cpunion commented:
默认 fork 会关闭 issues，设置打开即可。

flyskywhy committed:
Merge pull request #5 from cpunion/animated-scroll

ScrollView scrolls with animation

flyskywhy committed:
Merge pull request #6 from cpunion/scrollview-scrollenabled-default-true

Make ScrollView.defaultProps.scrollEnabled default true

flyskywhy commented:
@fov42550564 ，https://github.com/flyskywhy/react-web 中的 Issue 现在开启了

cpunion committed:
Add TextInput.State

cpunion committed:
Add storybook

cpunion committed:
Upgrade ref to new usage

flyskywhy committed:
Merge pull request #8 from cpunion/add-textinput-state

Add TextInput.State

flyskywhy committed:
Merge pull request #9 from cpunion/add-storybook

Add storybook

flyskywhy committed:
Update ListView.web.js

flyskywhy committed:
Update Picker.web.js

cpunion committed:
Remove style on HTML and BODY

cpunion committed:
Patch getLayout

flyskywhy committed:
just refactor

flyskywhy committed:
typo

yuzhiyi committed:
解决 Alert 没有 message 问题

flyskywhy committed:
Merge pull request #11 from yuzhiyi/master

解决 Alert 没有 message 问题

yuzhiyi committed:
Update extendProperties.web.js

flyskywhy committed:
Merge pull request #12 from yuzhiyi/patch-1

Update extendProperties.web.js

cpunion commented:
https://github.com/fmoo/react-native-web/commit/8a0da1f6aeecf2b772a20a5f154c206a006df234 这里有react-16 的 injectResponderEventPlugin 实现，可以借过来

cpunion committed:
Compatible with react/react-dom@^16.0.0-alpha.7

cpunion committed:
Compatible with react/react-dom@^16.0.0-alpha.7

cpunion committed:
Compatible demo

cpunion committed:
Compatible demo

cpunion committed:
Use prop-types, create-react-class in demo

cpunion committed:
Use prop-types, create-react-class in demo

flyskywhy committed:
support pointerEvents to work with react-navigation DrawerNavigator

flyskywhy committed:
Fix `Uncaught TypeError: Cannot assign to read only property 'pointerEvents' of object '#<Object>'` when HMR from react-hot-loader

cpunion committed:
Compatible with react-native EventEmitter path

cpunion committed:
Compatible with react-native EventEmitter path

cpunion committed:
Add EventEmitter.removeListener

cpunion committed:
Add EventEmitter.removeListener

cpunion committed:
Upgrade react-mixin to avoid assigning this.state in componentWillMount

flyskywhy committed:
Remove ^ in package.json as usual to controll version in node_modules/

flyskywhy committed:
Merge pull request #15 from cpunion/upgrade-react-mixin

Upgrade react-mixin to avoid assigning this.state in componentWillMount

flyskywhy committed:
Let ViewPager support not just array of View but also 1 View

so that it can support react-native-swiper with 1 View now

flyskywhy committed:
Support U4 core UC Browser

flyskywhy committed:
Support Selenium and Appium test with locator ~ (accessibilityLabel) and

    https://github.com/flyskywhy/CodeceptJS/commit/6d5fc76fdef9960de3f3e746a538a63c300e0bae
    https://github.com/flyskywhy/CodeceptJS/commit/d2b5215a6df7641b57a3dce63bdcdc80567a0a9c

flyskywhy committed:
Let ViewPager response width change to browser window size change

cpunion committed:
Allow React 16

cpunion committed:
Add addEventListener, removeEventListener to Dimensions.

flyskywhy committed:
Suppress warning by deprecated isMounted()

flyskywhy committed:
Let Text can measure() real height

killdahehe committed:
Update getLayout.web.js

修复web端onLayout返回的坐标是相对于根组件而不是父组件的问题

killdahehe committed:
Merge pull request #1 from killdahehe/killdahehe

Update getLayout.web.js

flyskywhy committed:
Merge pull request #17 from killdahehe/master

 Update getLayout.web.js

yuzhiyi committed:
Merge pull request #1 from flyskywhy/master

更新最新react-web

Commit:
TextInput增加setNativeProps功能

flyskywhy committed:
Merge pull request #18 from yuzhiyi/master

 TextInput增加setNativeProps功能

flyskywhy committed:
Fix `Uncaught TypeError: Cannot set property componentDidMount of #<TextInput> which has only a getter`

flyskywhy committed:
suppress Warning: prop type `textStyle` is invalid

flyskywhy committed:
suppress Warning: Unknown prop `source` on <img> tag

flyskywhy committed:
suppress Warning: Unknown props (many) on (<div> and <span>) tag

flyskywhy committed:
suppress Warning: Failed prop type: Invalid prop `source` supplied to `Image`

flyskywhy committed:
suppress Warning: prop type `iconStyle` is invalid when use Image.propTypes.style

flyskywhy committed:
suppress Warning: Unknown prop `testID` on <input> tag

flyskywhy committed:
suppress Warning: Unknown props `maxRows`, `minRows` on <textarea> tag

flyskywhy committed:
suppress Warning: Unknown prop `resizeMode` on <img> tag

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag

flyskywhy committed:
canOpenURL() need return Promise

flyskywhy committed:
BackHandler

flyskywhy committed:
react-dom: 16.0.0-alpha.2 to work with react@16.0.0-alpha.6 from react-native@0.43.0, and the reason
we set react-dom@16.0.0-alpha.2 but not react-dom@16.0.0-alpha.6 is:

1. react@16.0.0-alpha.6 -> ReactBaseClasses.js -> setState -> enqueueSetState()
is not compatible with the parameters defined in react-dom@15.4.2 ->
ReactUpdateQueue.js -> enqueueSetState(), so we need upgrade react-dom, othewise
navigator can't route to new page because it use setState;

2. ReactAnimated use animated, but animated has not suppor React v16 for now, see
https://github.com/animatedjs/animated/issues/63 , and react-dom@16.0.0-alpha.3
to react-dom@16.0.0-alpha.6 will crash the animated but accurate
react-dom@16.0.0-alpha.2 is good.

flyskywhy committed:
fix emitter.removeListener() ref to https://github.com/facebook/react-native/blob/405c573e0d41cdcb9f05b5a1b9a81a4ad9f97369/Libraries/EventEmitter/EventEmitter.js#L204

flyskywhy committed:
ViewPropTypes

flyskywhy committed:
ImageBackground

flyskywhy committed:
WebView

flyskywhy committed:
fix `Module not found: Error: Cannot resolve module 'react/lib/ReactMount'` when `react-web start`

flyskywhy committed:
Merge branch 'master' into tap-events

flyskywhy committed:
Merge pull request #13 from cpunion/tap-events

Compatible with react/react-dom@^16.0.0-alpha.7

flyskywhy committed:
Merge pull request #14 from cpunion/add-missing-functions

Add function EventEmitter.removeListener, compatible react-native EventEmitter path

flyskywhy committed:
Merge branch 'cpunion-merged-without-lib'

flyskywhy committed:
FlatList & List: upgrade from react-native@0.50.4

flyskywhy committed:
FlatList: upgrade from react-native@0.51.0

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag comes from react-native-swipeable

flyskywhy committed:
support Keyboard

yuzhiyi committed:
Merge pull request #2 from flyskywhy/master

new react-web

Commit:
修复Platform中select存在的bug

flyskywhy committed:
Merge pull request #19 from yuzhiyi/master

修复Platform中select存在的bug

Commit:
suppress Warning:Unknown props on FlatList

flyskywhy committed:
Merge pull request #20 from yuzhiyi/master

suppress Warning:Unknown props on FlatList

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag

yuzhiyi committed:
Merge pull request #3 from flyskywhy/master

update

Commit:
support onPress WebView

flyskywhy committed:
Merge pull request #21 from yuzhiyi/master

support onPress WebView

flyskywhy committed:
Animated(step 1): just copy from react-native@0.51.0

flyskywhy committed:
normalizeColor: just copy from react-native@0.51.0

flyskywhy committed:
Animated(step 2): worked well

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag

yucheng1207 committed:
Update ScrollView.web.js

fix scrollEventThrottle bug

flyskywhy committed:
Merge pull request #22 from yucheng1207/master

Update ScrollView.web.js

Commit:
AsyncStorage:worked well

yuzhiyi committed:
Merge pull request #4 from flyskywhy/master

merge new code

flyskywhy committed:
Merge pull request #23 from yuzhiyi/master

AsyncStorage:worked well

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag

flyskywhy committed:
suppress Warning: use ... instead of delete

flyskywhy committed:
suppress Warning: Unknown props on <span> tag

flyskywhy committed:
suppress Warning: Unknown event handler property `onComponentRef` on <div> tag

flyskywhy committed:
ImageBackground: fix width and height lost when style is not flatten

flyskywhy committed:
Animated: fix animation is stopped immediately by `useNativeDriver: true` even after warning "Falling back to JS-based animation"

flyskywhy committed:
typo

flyskywhy committed:
suppress Warning: Unknown props (many) on <div> tag

flyskywhy committed:
Picker: add selectedIndex to this.props.onValueChange() to work with https://github.com/ElekenAgency/ReactNativeWheelPicker

flyskywhy committed:
ListView: fix crash

flyskywhy committed:
ScrollView: support showsHorizontalScrollIndicator and showsVerticalScrollIndicator; 'auto' is better than 'scroll'

flyskywhy committed:
ScrollView: revert b364c3a with 1 and 2 in https://github.com/taofed/react-web/pull/102 , because
    1. if still position: absolute , problem 1 in https://github.com/taofed/react-web/pull/101 still exist
    when use ScrollView with DrawerItems of react-navigation even manually specify the height of ScrollView
    2. this padding workaround is the result of  position: absolute, so revert it too

flyskywhy committed:
WebView: use StyleSheet.flatten() to fix `Uncaught TypeError: Failed to set an indexed property on 'CSSStyleDeclaration': Index property setter is not supported.`

flyskywhy committed:
PanResponder: react-web work with react-dom 16.5 and newer comes from react-native@0.59.10 for [Expose less internals for TestUtils](https://github.com/facebook/react/pull/13539)

flyskywhy committed:
suggest use measureInWindow() instead of measure()
